finagle/buoyant/src/main/scala/com/twitter/finagle/buoyant/PathMatcher.scala

```diff
@@ -75,18 +75,29 @@ object PathMatcher {
             _extract(pathSegments.tail, exprSegments.tail, vars)
           else if (exprSegment == pathSegment)
             _extract(pathSegments.tail, exprSegments.tail, vars)
-          else if (exprSegment.startsWith("{") && exprSegment.endsWith("}"))
-            _extract(
-              pathSegments.tail,
-              exprSegments.tail,
-              vars + (exprSegment.substring(1, exprSegment.length - 1) -> pathSegment)
-            )
           else
-            None
+            """\{([a-zA-Z0-9\\.:-]+)\}""".r.findAllIn(exprSegment) match {
```

Comment:
I think it would be clearer to factor this code so that it happens when the Matcher is constructed, instead of at _extract time.  I'm thinking something like this:

When the Matcher is constructed and the exprSegments are split out, each one is stored as a Regex and a list of capture names.  For example, `/{foo}-{bar}` is stored as 

```
Seq(
  ExprSegment(regex = "(.*)-(.*)".r, variables = Seq("foo", "bar"))
)
```

This lets us compile all of the regexs ahead of time when the matcher is constructed.

Reply:
I see what you are saying ... this makes a lot of sense to me.  Thanks for the feedback.